From ce002f49f8783123cda113718bcc5ecfdc820710 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:39 -0400
Subject: [fs] btrfs: reserve no transaction units in btrfs_ioctl_set_features

Message-id: <1407179109-13714-153-git-send-email-sandeen@redhat.com>
Patchwork-id: 87190
O-Subject: [RHEL7.1 PATCH 152/422] btrfs: reserve no transaction units in btrfs_ioctl_set_features
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: David Sterba <dsterba@suse.cz>

commit 8051aa1a3d5aaa7bd4c062cad94d09c3d567ef2e
Author: David Sterba <dsterba@suse.cz>
Date:   Fri Feb 7 14:34:04 2014 +0100

    btrfs: reserve no transaction units in btrfs_ioctl_set_features

    Added in patch "btrfs: add ioctls to query/change feature bits online"
    modifications to superblock don't need to reserve metadata blocks when
    starting a transaction.

    Signed-off-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index 8e48b81..383ab45 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -4668,7 +4668,7 @@ static int btrfs_ioctl_set_features(struct file *file, void __user *arg)
  if (ret)
   return ret;
 
- trans = btrfs_start_transaction(root, 1);
+ trans = btrfs_start_transaction(root, 0);
  if (IS_ERR(trans))
   return PTR_ERR(trans);
 
-- 
1.7.1